frontend/src/components/modals/IntegrationOauthModal.tsx

```diff
@@ -65,38 +63,43 @@ export const OauthModal: React.FC<OauthModalProps> = ({
     };
 
     getOAuthURL();
-  }, [currentConfiguration, roadmapId, name]);
+  }, [currentConfiguration, roadmapId, name, retry]);
 
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     const form = event.currentTarget;
     event.preventDefault();
     event.stopPropagation();
 
-    if (form.checkValidity()) {
+    if (form.checkValidity() && currentConfiguration) {
       setIsLoading(true);
 
       const swapToken = async () => {
-        const success = await api.swapIntegrationOAuthToken(
-          name,
-          {
-            id: currentConfiguration.id,
-            verifierToken: formValues.oauthVerifierCode,
-            token: formValues.token,
-            tokenSecret: formValues.tokenSecret,
-          },
-          roadmapId,
-        );
-        setIsLoading(false);
-
-        if (success) {
-          closeModal();
-        } else {
+        try {
+          const success = await api.swapIntegrationOAuthToken(
+            name,
+            {
+              id: currentConfiguration.id,
+              verifierToken: formValues.oauthVerifierCode,
+              token: formValues.token,
+              tokenSecret: formValues.tokenSecret,
+            },
+            roadmapId,
+          );
+          if (success) {
+            closeModal(true);
+            return;
+          }
           setErrorMessage(
             `Unable to swap ${titleCase(
               name,
             )} OAuth token. Please contact an administrator if the problem persists.`,
```

Comment:
Translations to the error messages in IntegrationOauthModal could be added.
Here's how to add translations with variables:
https://www.i18next.com/translation-function/interpolation#working-with-data-models